content/en/docs/tasks/administer-cluster/memory-manager.md

```diff
@@ -0,0 +1,390 @@
+
+This error typically occurs in the following situations:
+* a node has not enough resources available to satisfy the pod's request
+* the pod's request is rejected due to particular Topology Manager policy constraints 
+
+The error appears in the status of a pod:
+```
+# kubectl get pods
+NAME         READY   STATUS                  RESTARTS   AGE
+guaranteed   0/1     TopologyAffinityError   0          113s
+```
+
+Use `kubectl describe pod <id>` or `kubectl get events` to obtain detailed error message:
+```
+Warning  TopologyAffinityError  10m   kubelet, dell8  Resources cannot be allocated with Topology locality
+```
+
+### System logs
+
+Investigate system logs with respect to a particular pod:
+
+```
+# journalctl -u kubelet | grep 821c5b92-a74f-4677-aa7d-acfbf70287ab
+```
+
+Sample output:
+```
+Nov 17 10:57:41 dell8 kubelet[52570]: I1117 10:57:41.086255   52570 topology_manager.go:198] [topologymanager] TopologyHints for pod 'guaranteed_default(821c5b92-a74f-4677-aa7d-acfbf70287ab)', container 'guaranteed': map[cpu:[{01 true} {10 true} {11 false}]]
+Nov 17 10:57:41 dell8 kubelet[52570]: I1117 10:57:41.086311   52570 topology_manager.go:198] [topologymanager] TopologyHints for pod 'guaranteed_default(821c5b92-a74f-4677-aa7d-acfbf70287ab)', container 'guaranteed': map[memory:[{11 true}]]
+```
+
+The set of hints that Memory Manager generated for the pod is shown in the output, i.e. `map[memory:[{11 true}]]`. Also, the set of hints generated by CPU Manager is present, i.e. `map[cpu:[{01 true} {10 true} {11 false}]]`.
+
+Topology Manager merges these hint to calculate single best hint shown below.
+
+Sample output:
+```
+Nov 18 10:41:22 dell8 kubelet[7227]: I1118 10:41:22.451426    7227 scope_container.go:53] [topologymanager] Best TopologyHint for (pod: gpod_default(2a97a5c6-0577-4956-9f61-153b735e95f6) container: gpod): {01 true}
+```
+
+The best hint (`{01 true}`) indicates where to allocate all the resources. Topology Manager tests this hint against its current policy, and based on the verdict, it either admits the pod to the node or rejects it.  
+
+Also, the grep might be used to search for occurrence of `[memorymanager]` in the logs:
+```
+Nov 03 13:23:00 dell8 kubelet[48363]: I1103 13:23:00.347272   48363 memory_manager.go:204] [memorymanager] Set container "046927bb304a02c1969d6389b28c6f279069e4d91bf6f726e028d714c06db0b1" cpuset.mems to "0"
+```
+
+### State file
+
+Let us first deploy a sample `Guaranteed` pod whose specification is as follows:
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: guaranteed
+spec:
+  containers:
+  - name: guaranteed
+    image: consumer
+    imagePullPolicy: Never
+    resources:
+      limits:
+        cpu: "2"
+        memory: 150Gi
+      requests:
+        cpu: "2"
+        memory: 150Gi
+    command: ["sleep","infinity"]
+``` 
+
+Next, let us log into the node where it was deployed and examine the state file in `/var/lib/kubelet/memory_manager_state`:
+```

```

Comment:
If you change this to <tt>\`\`\`json</tt>, the renderer adds syntax highlighting.